### PR TITLE
Add API/Monitoring node info to node list/status + fix certain bugs/nits in node cmds

### PR DESCRIPTION
**Pull request by felipemadero**
Bugs/Nits:
- separate monitogin node was not beign removed if created in devnet
- nodes non beign removed if trying to create nodes and the addrs limit is full
- outdated credentials msg not beign show if calling checkRegions
- if not elastic addresses were generated for addrs limit, recovery code anyway tried to release elastic addresses
- IP was not saved on node config for separate monitoring node if using dynamic IPs

### CODE REVIEW

File: cmd/nodecmd/list.go [hunk 1]
```diff
@@ -39,29 +41,41 @@ func list(_ *cobra.Command, _ []string) error {
...
-				return err
+			nodeIDStr := "----------------------------------------"
+			funcs := []string{}
+			if clusterConf.MonitoringInstance != cloudID {
+				nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
+				if err != nil {
+					return err
+				}
+				nodeIDStr = nodeID.String()
+				if clusterConf.IsAPIHost(cloudID) {
```
Comment (arturrez) [inline cmd/nodecmd/list.go:65]: nit: abstract adding `API` `Node` and `Monitor` to some function, so output is consistent everywhere

File: cmd/nodecmd/ssh.go [hunk 2]
```diff
@@ -75,6 +77,11 @@ func sshNode(_ *cobra.Command, args []string) error {
 			// try to detect nodeID
 			for clusterName := range clustersConfig.Clusters {
 				clusterHosts, _ := ansible.GetInventoryFromAnsibleInventoryFile(app.GetAnsibleInventoryDirPath(clusterName))
+				monitoringInventoryPath := filepath.Join(app.GetAnsibleInventoryDirPath(clusterName), constants.MonitoringDir)
```
Comment (arturrez) [inline cmd/nodecmd/ssh.go:80]: why do we want to include monitor host to ssh command?
are yiou trying to allow ssh command execution for hostID for monitoring ?

File: cmd/nodecmd/status.go [hunk 3]
```diff
@@ -234,46 +232,95 @@ func printOutput(
...
+				}
 			}
 			row = append(row, syncedStatus)
 		}
 		table.Append(row)
 	}
 	table.Render()
 }
+
+func removeColors(s string) string {
```
Comment (arturrez) [inline cmd/nodecmd/status.go:300]: you can use https://pkg.go.dev/github.com/pborman/ansi#Strip for this.
no need to hardcode list of colours

File: pkg/cloud/aws/aws.go [hunk 4]
```diff
@@ -309,23 +309,27 @@ func (c *AwsCloud) StopInstance(instanceID, publicIP string, releasePublicIP boo
...
-		if len(addressOutput.Addresses) == 0 {
-			return ErrNoAddressFound
-		}
-		releaseAddressInput := &ec2.ReleaseAddressInput{
-			AllocationId: aws.String(*addressOutput.Addresses[0].AllocationId),
-		}
-		if _, err = c.ec2Client.ReleaseAddress(c.ctx, releaseAddressInput); err != nil {
-			return err
+		if publicIP == "" {
+			ux.Logger.PrintToUser("Unabled to remove public IP for instannce %s: undefined", instanceID)
```
Comment (arturrez) [inline pkg/cloud/aws/aws.go:313]: nit: use ux.Logger.RedXToUser

File: cmd/nodecmd/list.go [hunk 5]
```diff
@@ -39,29 +41,41 @@ func list(_ *cobra.Command, _ []string) error {
...
+				} else {
+					funcs = append(funcs, "Node")
+				}
+			}
+			if nodeConfig.IsMonitor {
+				funcs = append(funcs, "Monitor")
+			}
+			funcDesc := strings.Join(funcs, ",")
+			if funcDesc != "" {
+				funcDesc = " [" + funcDesc + "]"
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/nodecmd/list.go:76]: I will just abstract the list of roles

File: cmd/nodecmd/ssh.go (hunk 2 above)
Comment (felipemadero) [reply cmd/nodecmd/ssh.go:80]: yes. I am adding that so can use ssh command also with the monitoring instance ip, cloud id, and whatever

Comment (felipemadero) [reply cmd/nodecmd/ssh.go:80]: this command should give access to all resources

File: cmd/nodecmd/ssh.go [hunk 6]
```diff
@@ -69,12 +74,31 @@ func sshNode(_ *cobra.Command, args []string) error {
 				if err != nil {
 					return err
 				}
+				monitoringInventoryPath := filepath.Join(app.GetAnsibleInventoryDirPath(clusterNameOrNodeID), constants.MonitoringDir)
```
Comment (arturrez) [inline cmd/nodecmd/ssh.go:77]: we should not allow running ssh commands for the cluster on monitoring instance as it's very different.  Everything else is the same so it makes sense to allow it as all boxes are deployed exactly the same so there is value in running one command across cluster.
For monitoring intance as it's not running avalanchego - it should be excluded from cluster wide ssh commands.

But I still see value allowing users to reference monitoring instance per cloud instance ID to run ssh command or open terminal sessions

**Review by arturrez — COMMENTED** [review]
https://github.com/ava-labs/avalanche-cli/pull/1538/files#r1515124123

**Review by arturrez — APPROVED** [review]

**Pull request merged**